collate/collate.py

```diff
@@ -197,9 +197,9 @@ def get_columns(self, when=None, prefix=None, format_kwargs=None):
             yield ex.literal_column(column).label(to_sql_name(name))
 
 
-def maybequote(elt):
+def maybequote(elt, quote_choices):
     "Quote for passing to SQL if necessary, based upon the python type"
-    if isinstance(elt, Number):
+    if quote_choices is False or (quote_choices is None and isinstance(elt, Number)):
```

Comment:
It'd be nice to separate these behaviors a bit more clearly — I'm thinking something like

```
if `always_quote` is None:
    # do the smart-quoting
elif `always_quote`:
    # just put quotes around it
else:
    # return the element
```